pkg/scaling/resolver/gcp_secretmanager_handler.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/crc32"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"github.com/go-logr/logr"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+type GCPSecretManagerHandler struct {
+	gcpSecretsManager       *kedav1alpha1.GCPSecretManager
+	gcpSecretsManagerClient *secretmanager.Client
+	ctx                     context.Context
```

Comment:
I'm also thinking that maybe now with the 3rd secret provider is a good moment to refactor a bit the approach and creating an interface for them. I mean, something like:

```golang
type ExternalSecretsManager interface {
	Read(ctx context.Context, secretName string, version string)
	Initialize(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, secretsLister corev1listers.SecretLister) error
}
```
It allows unifying the behavior with other secret managers. WDYT? @zroubalik ?
We aim to do a release on next week, so we can just keep this (just applying the context change) and opening an issue to track the unification

Reply:
WDYT about this?

Reply:
I can take this up, if you'd like

Reply:
I think it's a good improvement to standardize the things because we already have 3 different implementations and at least one more will be eventually added (aws kms)

Reply:
I think we can proceed @neelanjan00

Reply:
absolutely +1 for standardization

Reply:
@JorTurFer I think we should add these changes collectively in a separate PR?

Reply:
Do you agree @zroubalik ? I guess we can do it as part of another PR

Reply:
Yes, let's do it in a separate PR.